### ISSUE TITLE
{badmatch,{error,closed}} in pgo_connection

### ISSUE THREAD

**Issue by benbro** [erleans/pgo#42]
To reproduce the error, restart DB server on a different machine.
pgo_handler:ping(Conn) in line 123 should handle {error, closed} response?

```erlang
handle_event(cast, {ping, Holder}, Data=#data{pool=Pool,
                                              holder=Holder,
                                              queue=QueueTid,
                                              conn=Conn}) ->
    ok = pgo_handler:ping(Conn),
    NewHolder = pgo_pool:update(Pool, QueueTid, ?MODULE, Conn),
    {keep_state, Data#data{holder=NewHolder}};
```
Log:
```erlang
supervisor: {<0.724.0>,pgo_connection_sup}
errorContext: child_terminated
reason: {{badmatch,{error,closed}},
         [{pgo_connection,handle_event,3,
                          [{file,"pgo/src/pgo_connection.erl"},
                           {line,123}]},
          {gen_statem,loop_state_callback,11,
                      [{file,"gen_statem.erl"},{line,1166}]},
          {proc_lib,init_p_do_apply,3,
                    [{file,"proc_lib.erl"},{line,226}]}]}
offender: [{pid,<0.838.0>},
           {id,pgo_connection},
           {mfargs,
               {pgo_connection,start_link,
                   [#Ref<0.1330457315.1750590656.121213>,<0.233.0>,
                    default,<0.824.0>,
                    #{database => "app",
                      host => "localhost",
                      password =>
                          "*******",
                      pool_size => 5,ssl => true,
                      ssl_options => [{verify,verify_none}],
                      user => "app"}]}},
           {restart_type,permanent},
           {shutdown,100},
           {child_type,worker}]
```

**Comment by tsloughter** [comment]
To be clear, does everything work ok even though it outputs this log?

Handling the message so it can log a better message about what happened would be great, just making sure that the crash itself doesn't hurt anything but the usability of the logs.

**Comment by benbro** [comment]
I didn't see a problem. I think the client automatically reconnected.
Logging a short message instead of a long crash report will be much better.

**Comment by tsloughter** [comment]
Cool, yea, it was meant to just be a quick way to force it to crash if connection doesn't work and then it brings up a new one, but handling it, logging an info message and calling `close_and_reopen` would be a good update.

**Comment by benbro** [comment]
I saw you are planning a new release. Any chance we could have this issue and https://github.com/erleans/pgo/issues/41 fixed before?

**Comment by tsloughter** [comment]
Ah yea, I can try to get fixes for these figured out before 1.0.0-rc1

**Comment by tsloughter** [comment]
Resolved by #58
